upload-chunked-file.js

```diff
@@ -0,0 +1,43 @@
+function chunkFileUpload(file, serverUrl, chunkSize) {
+    const totalChunks = Math.ceil(file.size / chunkSize);
+    let chunkIndex = 0;
+
+    (function uploadNextChunk() {
+        const start = chunkIndex * chunkSize;
+        const end = Math.min(start + chunkSize, file.size);
```

Comment:
Loading and setting headers on a new `XMLHttpRequest` for each chunk is inefficient. Consider reusing a single `XMLHttpRequest` object and only updating its attributes as necessary.